src/app/pages/Collectibles/CollectiblesList.tsx

```diff
@@ -33,12 +39,40 @@ const CollectiblesList = () => {
         slugs.push(tokenSlug);
       }
     }
+    canLoadMore.current = true;
 
     return slugs;
   }, [collectibles, allTokensBaseMetadata]);
 
   const { filteredAssets, searchValue, setSearchValue } = useFilteredAssets(assetSlugs);
 
+  const handleLoadItems = useCallback(() => {
+    if (canLoadMore.current) {
+      canLoadMore.current = false;
+      loadItems();
+    }
+  }, [loadItems]);
+
+  useEffect(() => {
+    const el = loadMoreRef.current;
+    if ('IntersectionObserver' in window && el) {
+      const observer = new IntersectionObserver(
```

Comment:
a lot of similar logic with `IntersectionObserver`.  
extract common code into shared hook